src/pybind/mgr/rgw/cli.py

```diff
@@ -0,0 +1,221 @@
+        parser.add_argument('--no-start-radosgw', action='store_false', dest='start_radosgw')
+
+        args = parser.parse_args(self.args[1:])
+
+        return RGWAM(self.env).zone_create(args.realm_token, args.zonegroup, args.zone, args.endpoints, args.start_radosgw)
+
+class CommonArgs:
+    def __init__(self, ns):
+        self.conf_path = ns.conf_path
+        self.ceph_name = ns.ceph_name
+        self.ceph_keyring = ns.ceph_keyring
+
+class TopLevelCommand:
+
+    def _parse(self):
+        parser = argparse.ArgumentParser(
+            description='RGW assist for multisite tool',
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog='''
+The commands are:
+   realm bootstrap               Bootstrap new realm
+   realm new-zone-creds          Create credentials to connect new zone to realm
+   zone create                   Create new zone and connect it to existing realm
+''')
+
+        parser.add_argument('command', help='command to run', default=None)
+        parser.add_argument('-c', help='ceph conf path', dest='conf_path')
+        parser.add_argument('-n', help='ceph user name', dest='ceph_name')
+        parser.add_argument('-k', help='ceph keyring', dest='ceph_keyring')
+
+        removed_args = []
+
+        args = sys.argv[1:]
+        if len(args) > 0:
+            if hasattr(self, args[0]):
+                # remove -h/--help if top command is not empty so that top level help
+                # doesn't override subcommand, we'll add it later
+                help_args = [ '-h', '--help' ]
+                removed_args = [arg for arg in args if arg in help_args]
+                args = [arg for arg in args if arg not in help_args]
+
+        (ns, args) = parser.parse_known_args(args)
+        if not hasattr(self, ns.command) or ns.command[0] == '_':
+            print('Unrecognized command:', ns.command)
+            parser.print_help()
+            exit(1)
+        # use dispatch pattern to invoke method with same name
+        args += removed_args
+        return (getattr(self, ns.command), CommonArgs(ns), args)
+
+    def realm(self, env, args):
+        cmd = RealmCommand(env, args).parse()
+        return cmd()
+
+    def zone(self, env, args):
+        cmd = ZoneCommand(env, args).parse()
+        return cmd()
+
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+
+    log = logging.getLogger(__name__)
+
+    (cmd, common_args, args)= TopLevelCommand()._parse()
+
+    env = EnvArgs(RGWAMCLIMgr(),
+                  common_args.conf_path,
+                  common_args.ceph_name,
+                  common_args.ceph_keyring)
+
+    try:
+        retval, out, err = cmd(env, args)
+        if retval != 0:
+            log.error('stdout: '+ out + '\nstderr: ' + err)
+            sys.exit(retval)
+    except RGWAMException as e:
+        print('ERROR: ' + e.message)
+
+    sys.exit(0)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Moved code around, see: https://github.com/ceph/ceph/pull/42710/commits/0eb108fab37671f35ae1931eb891a6014b7c509b. Doesn't deploy the standalone cli command as there were some concerns about adding this while we want to move users to use cephadm stuff. It can still be used for dev work.